cloud-controller-manager/do/tags.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2017 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package do
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/digitalocean/godo"
+	"github.com/golang/glog"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	v1informers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	v1lister "k8s.io/client-go/listers/core/v1"
+)
+
+const (
+	controllerName       = "tagscontroller"
+	tagsLabelPrefix      = "digitalocean.com/droplet-tag/"
+	controllerSyncPeriod = time.Minute
+)
+
+type tagsController struct {
+	kclient    kubernetes.Interface
+	gclient    *godo.Client
+	nodeLister v1lister.NodeLister
+}
+
+// NewTagsController returns a new tags controller responsible for applying
+// droplet tags to node labels. It will return an error if the kubernetes client
+// failed to initialize.
+func NewTagsController(n v1informers.NodeInformer,
+	k kubernetes.Interface, g *godo.Client) *tagsController {
+	return &tagsController{
+		kclient:    k,
+		gclient:    g,
+		nodeLister: n.Lister(),
+	}
+}
+
+// Run starts the tags controller. Tags controllers runs on a loop rather than watching
+// for changes because there is no way to watch for droplet events (yet) and watching
+// node events can be expensive as each node update event (mostly node health checks)
+// would result in an API call to fetch droplets. Caching droplets in memory
+// may be a better solution but this should work for now.
+// TODO: pass in stopch once it's supported upstream
+func (t *tagsController) Run() error {
+	tick := time.Tick(controllerSyncPeriod)
```

Comment:
Can you use `time.NewTicker()` and call `ticker.Stop()` in a defer call. This is open door for goroutine leaks.